src/main/Extensions/SimpleFolderSearch/SimpleFolderSearch.ts

```diff
@@ -0,0 +1,121 @@
+
+export class SimpleFolderSearch implements Extension {
+    public readonly id = "SimpleFolderSearch";
+    public readonly name = "Simple Folder Search";
+
+    public readonly author = {
+        name: "Sasha Weiss",
+        githubUserName: "sashaweiss",
+    };
+
+    private readonly defaultSettings: Record<OperatingSystem, { folders: string[] }> = {
+        Windows: {
+            folders: [this.app.getPath("home")],
+        },
+        macOS: {
+            folders: [this.app.getPath("home")],
+        },
+        Linux: {
+            folders: [this.app.getPath("home")],
+        },
+    };
+
+    private readonly folderIconImage: Image = {
+        url: `file://${this.assetPathResolver.getExtensionAssetPath(this.id, "macos-folder-icon.png")}`,
+    };
+
+    public constructor(
+        private readonly app: App,
+        private readonly operatingSystem: OperatingSystem,
+        private readonly assetPathResolver: AssetPathResolver,
+        private readonly fileSystemUtility: FileSystemUtility,
+        private readonly settingsManager: SettingsManager,
+        private readonly translator: Translator,
+    ) {}
+
+    public async getSearchResultItems(): Promise<SearchResultItem[]> {
+        type SearchResult = {
+            fullPath: string;
+            abbreviatedName: string;
+            fileName: string;
+        };
+
+        const searchResultBatches: SearchResult[][] = await Promise.all(
+            this.getPathsToSearch().map(async (folderPath: string): Promise<SearchResult[]> => {
+                const directoryEntries = await this.fileSystemUtility.readDirectory(folderPath);
+                return directoryEntries.map((directoryEntry): SearchResult => {
+                    return {
+                        fullPath: directoryEntry,
+                        abbreviatedName: directoryEntry.replace(this.app.getPath("home"), "~"),
+                        fileName: path.parse(directoryEntry).base,
+                    };
+                });
+            }),
+        );
+
+        const searchResults: SearchResult[] = searchResultBatches.reduce(
+            (accumulator: SearchResult[], nextValue: SearchResult[]): SearchResult[] => {
+                return accumulator.concat(nextValue);
+            },
+            [],
+        );
+
+        // Filter out non-directories, and hidden items.
+        const filteredSearchResults: SearchResult[] = searchResults.filter((searchResult: SearchResult): boolean => {
+            return this.fileSystemUtility.isDirectory(searchResult.fullPath) && !searchResult.fileName.startsWith(".");
+        });
+
+        return filteredSearchResults.map((searchResult: SearchResult): SearchResultItem => {
+            return {
+                id: `${this.id}-path-${searchResult.fullPath}`, // Ensure unique
+                name: searchResult.fileName,
+                description: `Open ${searchResult.abbreviatedName}`,
+                image: this.folderIconImage,
```

Comment:
Instead of having this static image you can also use the `FileImageGenerator` from `@Core/ImageGenerator` to get more dynamic icons for folders and files.

Just add the FileImageGenerator to the constructor and use it like this:

``` typescript
const images = await this.fileImageGenerator.getImages(searchResults.map(({ fullPath }) => fullPath));

return filteredSearchResults.map((searchResult: SearchResult): SearchResultItem => {
    return {
        id: `${this.id}-path-${searchResult.fullPath}`, // Ensure unique
        name: searchResult.fileName,
        description: `Open ${searchResult.abbreviatedName}`,
        image: images[searchResult.fullPath] ?? this.folderIconImage, // <= use the extracted icon and fallback to the generic image
        defaultAction: SearchResultItemActionUtility.createOpenFileAction({
            filePath: searchResult.fullPath,
            description: `Open ${searchResult.abbreviatedName}`,
        }),
    };
});
```

Reply:
Happy to include this change if you'd prefer stylistically – I think there is some value too in using the same blanket icon for all results for the extension, to visually separate them from, for example, application search results.